### ISSUE TITLE
[BUG] Not working on arch budgie with budgie 10.5

### ISSUE THREAD

**Issue by botrosco** [dgaw/budgie-workspaces-compact#2]
Is this project still maintained if so would you be able to help me with this issue that im having.
Within arch (antergos) the plugin does not seem to work correctly, i am able to rightclick and set the number of workspaces however if clicking or using the scroll wheel while hovering over the applet nothing seems to happen. 
Also if i have the default workspace applet open the setting of the number of workspaces doesnt seem to change the number of available workspaces, i am unsure if this is indended or not. I have very little python experience so i dont really know what to look for in the code or if this is an issue that was casued by the move to budgie 10.5. 
I havent use this applet before so i am unaware if this ever worked on arch.

**Comment by dgaw** [comment]
I haven't used this applet in a while. However I just did a quick check and it seems to work fine on Solus 4.0 with budgie-desktop 10.5.

I'm not sure what might be wrong on Arch but if you feel like troubleshooting this then you can enable some debugging information in the applet by changing line:

`logging.basicConfig(level=logging.INFO)`

to 

`logging.basicConfig(level=logging.DEBUG)`

and restarting the panel with 

`budgie-panel --replace &`

On my machine this produces the following log when I switch workspaces with a left-click or scroll on the applet:

```
DEBUG:workspaces_compact:on_workspace_changed()
DEBUG:workspaces_compact:update_spin_button() ws count: 4
DEBUG:workspaces_compact:on_workspace_changed()
DEBUG:workspaces_compact:update_spin_button() ws count: 4
DEBUG:workspaces_compact:on_workspace_changed()
DEBUG:workspaces_compact:update_spin_button() ws count: 4
...
```

**Comment by botrosco** [comment]
Ok so it seems that the issue i am having is not changing workstations but adding them. If i add a workstation with the default applet i am able to change between the number of workstations that are present via that applet. However i am unable to do this with your applet and ther doesnt seem to be any errors in the logs.

https://youtu.be/MlvGE7ujJSc

Here is a video of what I am experiencing

**Comment by dgaw** [comment]
Interesting. Perhaps there's an unwanted interaction between the default applet and my applet. 

Did you try removing the default applet from your panel before changing the number of workspaces in the compact applet?

**Comment by botrosco** [comment]
I have yes and have had no luck.

**Comment by dgaw** [comment]
I see. I did a bit more digging and it seems the following bit of code doesn't work as expected on your system:

Line 111 in workspaces_compact.py:

`self.wn_screen.change_workspace_count(val)`

Unfortunately it works fine here and I don't have an Arch distro handy so that's as far as I can go.

Perhaps someone experiencing the same issue can look into it further.

**Comment by Jacob-Vlijm** [comment]
You might be missing Wnck as a dependency. @botrosco  @dgaw .
`python3-gi gir1.2-wnck-3.0` to be precise.

**Comment by botrosco** [comment]
Im not at my desktop rn so can't check atm but will give that a look when i can thanks.

**Comment by dgaw** [comment]
@Jacob-Vlijm Yes, it's worth checking that, although I think the applet would have crashed if the dependencies hadn't been there.

**Comment by botrosco** [comment]
wnck was allready installed, I installed the python bindings for wnck from AUR and still no luck.

**Comment by Jacob-Vlijm** [comment]
@dgaw Ah, silly, you are right of course, even the import would break the applet.
